### ISSUE TITLE
[FR] MegaProjects

### ISSUE THREAD

**Issue by OmegaHaxors** [Electrical-Age/ElectricalAge#880]
Megaprojects is a new way to play the game for the truly dedicated. Unlike normal play, a megaproject is a very large structure that provides a very large amount of power. Like the name implies, you can not benefit from a megaproject until every single part of it is built, so you must be willing to commit to it but the instant you complete it, you will benefit from great power outputs. However, for a megaproject to stay successful, regular maintenance should be put on it. Failure to do so may have catastrophic consequences, and may require you to completely rebuild! As long as you engineer everything properly, you shouldn't have to worry much about things failing often if at all.

### MegaTurbine:

The megaturbine is a very large wind turbine, similar to the ones you see in real life. These aren't your 50V turbines, no. They are the simplest megaproject and require the least maintenance. As an added bonus, they will occasionally kill birds, causing feathers and bird meat to fall. If allowed to over-spin, they will shear apart, forcing you to replace the blades. If you are active when one starts to overspin, you can take action to prevent this from happening, saving your turbine from failure.

![2018-08-29_16 07 42](https://user-images.githubusercontent.com/16057503/44820426-b9cd6780-aba5-11e8-9fe3-6ba627d9a0e2.png)

### MegaTower:

The megatower is to solar as the turbine is to wind. Since large-scale silicon-based panels are impractical for such a large project, heat is captured and beamed to a central location instead, where it heats up some fluid which can then be put through a turbine. If coolant isn't properly cycled, or if the tower exterior is not allowed to cool (from skipping nights) then the tower will eventually start to melt. If action isn't taken, then coolant may spill out, creating a large mess. The mirrors used to reflect the light to the center is also fragile and can easily be broken by players or NPCs stepping on them, however you may choose to reinforce the glass with lead.

### MegaDam:

The water turbine is really weak, but consistent. Now, what if you have a lot of time on your hands to terraform, and desire a stronger method of obtaining consistent power? You build a megadam, that's what! Out of all of the megaprojects, this one is by far the weakest. However, unlike other projects it requires no maintenance at all! This is balanced out by the very very very large amount of effort required to build one up front. First you have to dig out a very large area, the wider the area the more water you will gain when it rains, and the deeper the area the more water you can store. Next, you have to build a hydroturbine. The larger the width of the turbine and the more turbines you build, the more power you can extract out of the water. With a proper setup, you can extract 100% of the energy from the water! This allows it to double as a method for storing energy, by pumping the water back up for use later.

### MegaThermal:

Legends speak of a place filled with endless Lava, and while getting any machinery to stay cool enough to operate in there is a challenge in itself, you feel that with the right mix of science, engineering, and insanity you can make it happen. Now the question you have to ask yourself is how you are going to send the high-energy lava from the Nether to your waiting turbine setup? You use an experimental portal to send it to your home dimension, of course! While megathermal may be the strongest project up this point, it also has the most devastating impact on the environment, and requires nearly constant interactions while it runs. You will need to spend a lot of energy to keep the portal running, and you will need to constantly suck up lava to keep the entire thing a net positive. Furthermore, the portal is very unstable and won't tolerate collisions with the environment. If only there was a way to see where it was going to go... Thankfully at the very least you can control the thing remotely to make sure it stays within range of a lava pool, and with a little bit of outside help (or some automation) you can have some seeing-eyes to keep your portal away from danger. What are you going to do with all the power generated by this? Who knows! Science said you could, engineering says you should, and insanity says you already did.

### MegaReactor:

Megathermal taught you a lot, and one of the things it taught you is that very large volumes of power is addicting and that running multidimensional-ops can be tiring. While sucking up all that lava, you found some Uranium in the slag of the imported lava, and instantly know what to do with it. Time to make some atom bombs! Well, it's little brother at least, a fission reactor. It's time to put your engineering mettle to the ultimate test by going 'all in' on your base (and the immediate vicinity)'s ability to remain anything other than a pile of heavily irradiated ash. While your experiments with Uranium are all fine and dandy, it does make you wonder if there's even more. That's right, put on your hazard suits because we're going to be breeding some Plutonium. Even the admins are saying you're going too far, show them otherwise! Just watch out for nuclear waste. I heard that stuff is bad for the skeleton.

### MegaFabrication:

So you might have heard of this concept a really smart guy came up with. He predicted you would be able to convert matter into energy. You already knew that though, but have you ever stopped to think the process could be reversed? It would take a lot of power, a lot of equipment, and some pretty precise math but you could theoretically produce any item you know the composition of out of thin air. We are not responsible for any dirty looks you may get when you use this divine power to make yourself a tuna sandwich. (I was hungry, Ok?)

### MegaStellar:

Far enough is never far enough, and at this point you've exhausted any reason to play the game. You want to build something for the sake of building it, of being the best, of never having to build another thing again. You're sick of constantly dealing with nuclear waste and you just want some free and easy power, and you want lots of it. With a lot of effort, a lot of knowledge, and a PhD in at least 30 fields, you can combine everything you know to build the ultimate in human achievements: The fusion reactor. What could you possibly use all that power on? Well...

### MegaPortal:

This is it. You've done everything there possibly is to do. You've climbed the tallest mountains and built the largest bombs. You've warped across the galaxy using dubious crystals. You've bent the laws of nature to your will, and now you have nothing to show for it. Maybe it's better to just leave it all behind, and find a new world, where the rules aren't the same. Everybody said it was impossible, and your chances of surviving this thing are pretty slim if even a single digit of your math is off. It's time to become king not only of this world, not only of this universe, but of all of them. What could possibly go wrong? Who cares, go forth and claim your birthright.

**Comment by jrddunbr** [comment]
Hmm, well thought out.

**Comment by Baughn** [comment]
Most of these should be possible to implement using smaller building blocks, which I feel would be very much in tune with Eln. This is a good list to keep in mind when picking building blocks, though.

**Comment by Baughn** [comment]
Thought of the day: At somewhat higher tech levels, the workshop can have a 'CAD' mode that allows players to design turbines like the windmills. Even if it doesn't make sense to use multiple minecraft-level blocks, that still fits well into the parametric block system.

We can compute most of the behaviour statically, even with custom coil designs.

**Comment by AMIDIBOSS** [comment]
This all seems nice, but I have absolutely no idea how to model the dam. Let's start with the turbines after the 1.10 port. They're the easiest to implement...

**Comment by voidsong-dragonfly** [comment]
So are the “megaprojects”, if ever implemented, going to be multiblocks? Or one or two blocks? Or “Reika-style” multiblocks where they do not form, but all the parts just interact?

**Comment by jrddunbr** [comment]
Probably a mix. Depends on what people want.

**Comment by voidsong-dragonfly** [comment]
I am all in favor of the realistic mega-projects, especially if they are Reika-style mutliblocks where the mutliblock is all just parts that interact, and they then constitute the final structure. I, however, would like some other way of getting uranium for a fission reactor than the lava from the nether thing.

**Comment by OmegaHaxors** [comment]
I'm not 100% sure on the actual method used to make a MegaProject, but a few things that will be certain is that the size will be variable based on how many resources you use, and you will likely either require or have access to a machine that will help you build it. Think like a machine that will automatically assemble them but slowly or you could place the blocks yourself after milling them. As for uranium being super hard to get that's kind of the point. Too many mods have Uranium just strewn around the ground willy-nilly to the point where it's not even useful unless you build a stupidly oversized reactor. Maybe could allow Uranium to be bred using Thorium which could be extracted from the lava directly without any complicated tools for those who would rather just jump directly to it but even that's a bit of a stretch. Would also probably be worth renaming it Netherium and Rackium respectively so that it won't oredic with the actual elements they're based off of.

Megaprojects are supposed to be strong but not exactly practical compared to normal methods, and if uranium is too easy to obtain than it instantly steps on the toes of every other power source and becomes 'the only' generator. That's something I want to reserve to Fusion which will be so expensive to start and maintain that it becomes extremely unviable to ever use it unless you're super dedicated.

**Comment by voidsong-dragonfly** [comment]
I agree that some mods have too much uranium ore, however I like having it as ore. Though maybe restricting it to certain places [not to copy ReactorCraft, but Reika’s idea was good]. I would like having a realistic process for enriching uranium, which in itself is a task on its own, but if you enrixhed the uranium somehow that, say needs something on the megaproject tree before it (For example insane power requirements), that seems fine as well. I also like how ELN is mostly realistic, so would prefer keeping it as Uranium and Thorium.

**Comment by OmegaHaxors** [comment]
The reason I don't want to copy actual reactor systems is because of how many mods add them already. Reactors are already a vital part of Electrical Age's power generation through the use of the steam generator. Adding in an in-mod way to generate fission wouldn't really add anything if it was exactly the same as existing methods, which is why it has its own unique retrieval system. Instead of centrifuging uranium at great cost, you can get pure Nt238 from MegaThermal or you can breed up Rk234 using some small chain to get the stuff that way. Then the wastes produced from this would eventually decay down to useful products that can be used to make MegaFusion and a few of the more expensive machines. That way it serves an in-game purpose without stepping on existing methods.

**Comment by voidsong-dragonfly** [comment]
Yeah, and I get the “No more fission please”, as many mods already have it, however I still like realism. I like fission, but we don’t have much that is fissile other than what has already been mentioned.  Maybe curium or Neptunium, but those don’t seem to be viable reactor fuels, just fissile. Those could be used to do something, as neptunium, after what seems to be a million decays, decays into bismuth, and curium into plutonium... but then we just have an alternate way for plutonium, so maybe getting neptunium some way and then fission would eventually have it decay into bismuth which has some application later on. A couple of the elements could also be used for other small tasks. I would just prefer not having to make up elements to fission, but we could do it if we have to because the nether is strange and could have new things. Your ideas, so it is your call, however I would like a somewhat realistic way.

**Comment by OmegaHaxors** [comment]
I wouldn't be against using some lesser fissile materials for the sake of useful byproducts. It would be preferable to just making up fantasy materials and would also teach a non-uranium fuel chain.

Kind of like how minechem does it. The reactors are for making new materials, not for power.

**Comment by voidsong-dragonfly** [comment]
[]()https://periodic.lanl.gov/93.shtml - Neptunium is most commonly a fission byproduct of Uranium fission,  but it could fission itself. I cannot find what its fission byproducts are, so we could just use this decay chain: Np-> Pa -> Ac -> Fr -> Ra -> Rn -> [Lead, Bismuth, Polonium]. Neptunium can also be made into superconductors (top link). Curium decays into plutonium 238, for RTGs, and is also produced in conventional reactors as waste. Both use fast neutron reactors. Neptunium can be found in trace amounts in Uranium ores. The bismuth that you get could be used for heat-resistant alloys with iron and cadmium, but could also be supplemented with bismuth from processing lead, copper, or gold (If needed)  - []()https://education.jlab.org/itselemental/ele083.html .  As Pu238 is used in RTGs, we could also have it be used for irradiating lithium? Or maybe just have the neptunium reactor irradiate lithium with bismuth as an eventual byproduct and use that for heat-resistant alloys in the fusion reactor. The fusion reactor could also require the neptunium superconductors I mentioned above, but those need aluminum as well.

EDIT~ Also, I really like the idea of having MegaPortal go to some new dimension where thing are just subtly different (For example only ELN teleporters work so you cannot get back), but then closing behind you so you have what you take with you and only what you take with you, but it requires the fusion reactor's power. That way you let people either behave like they are starting a new world as you suggest, but also, if they are smart, let them take what they would need to start a new, already into electrical age base.

**Comment by voidsong-dragonfly** [comment]
FROM DISCORD:
Megaproject consumer #1
A TBM(Tunnel Boring Machine)-https://en.m.wikipedia.org/wiki/Tunnel_boring_machine
Note: This is all based on Double-Shield TBMs, as they seem the most applicable
Your goto for *fully* automated mining. With a constant input of concrete segments to line the tunnel walls, constant removal of debris, and large amounts of power, the machine will cut through rock with a tungsten (carbide)?grinding head, shoving out the ground up bits out the back of the grinding head. From these, you can refine ores, increasing your ore yield. The TBM will mine forever in whatever pattern you specify, as long as it can make the pattern (no sharp turns, etc). It needs *precise* automation to keep it going, as it needs spoil moved away, concrete segments delivered, etc. As with IRL, the better you have the system automated, the faster your TBM will go. If you run out of concrete, the walls will cave in. If you push the TBM too fast, your automation will not be able to keep up and fail, possibly catastrophically. If spoil is not removed from the head fast enough, buildup can cause failure, or breakage of the grinding head. Ajustable size of head between 3 and 17 block diameter, with larger machines consuming more power.
